### PR TITLE
adding z index to fix sidebar-page overlapping

### PR DESCRIPTION
**Pull request by techy4shri**
Modified the css files of card and sidebar to fix the overlapping of project card over sidebar with minor changes in the sidebar jsx file. 

The package.json file is modified for the version of vite used.

This does not affect the behavior of either element.

See #246

### Issue Title: Bug 🐛: The side menu gets covered by page item when a webpage opens #246

- **Name of the project** :  Adjusting the overlapping issue in a web page and side menu
- **Category:**  CSS, Front-end,React, JavaScript
- **Name:** GARIMA
- **GitHub ID:** techy4shri
- **Idenitfy yourself:**  GSSOC Contributor

Closes: 246

### Describe the add-ons or changes you've made 📃

Modified the css files of card and sidebar to fix the overlapping of project card over sidebar with minor changes in the sidebar jsx file. Now the sidebar comes in front when expanded and the project card stays at the back.

The package.json file is modified for the version of vite used.

This does not affect the behavior of either element.

See #246

### Type of change ☑️

What sort of change have you made:

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, local variables)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### How Has This Been Tested? ⚙️

I checked the changes by running the project. I went through all the pages, checked them by hiding and exapnding the sidebar menu and hovering over the adjacent project card.

### Checklist: ☑️

- [x] My code follows the [Contributing Guidelines](https://github.com/Avdhesh-Varshney/WebMasterLog/blob/main/CONTRIBUTING.md) & [Code of Conduct](https://github.com/Avdhesh-Varshney/WebMasterLog/blob/main/CODE_OF_CONDUCT.md) of this project.
- [x] This PR does not contain plagiarized content.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly wherever it was hard to understand.
- [x] My changes generate no new warnings.

### Screenshots 📷

BEFORE CHANGES:
Screenshot 2024-06-05 222623 1
AFTER CHANGES:
image

### Note to reviewers 📄
The sidebar menu comes in front when expanded, please check whether this is all that needs to be done or you would like me to add anything else.

### CODE REVIEW

File: src/components/Sidebar/sidebar.css
Comment (Avdhesh-Varshney) [inline src/components/Sidebar/sidebar.css:1]: format this file in 4-spacing as there are much more un-necessary changes of just formatting in 2-spacing.

**Review by Avdhesh-Varshney — CHANGES_REQUESTED** [review]
@techy4shri LGTM. but the un-necessary changes in the code makes hard to review the pr.

File: src/components/Sidebar/sidebar.css
Comment (techy4shri) [reply src/components/Sidebar/sidebar.css:1]: @Avdhesh-Varshney please check whetehr the changes reflect now and review.

**Review by Avdhesh-Varshney — CHANGES_REQUESTED** [review]
@techy4shri In this way, pr can't approved. Change that much code which is necessary not don't make me fool to commit by adding some spaces.

**Comment by techy4shri** [comment]
> @techy4shri In this way, pr can't approved. Change that much code which is necessary not don't make me fool to commit by adding some spaces.

No one is making you a fool here. You yourself told me to format the file in 4 space in the review so I did.

**Comment by Avdhesh-Varshney** [comment]
create another pr, this can't be approved. Add your code only don't change everything
image

**Comment by techy4shri** [comment]
Okay.

**Pull request closed without merging**